src/utils/report.rs

```diff
@@ -0,0 +1,71 @@
+use std::env;
+
+use crate::consts::OFF_CHAIN_AGENT_GRPC_URL;
+use leptos::{server, ServerFnError};
+
+pub enum ReportOption {
+    Nudity,
+    Violence,
+    Offensive,
+    Spam,
+    Other,
+}
+
+impl ReportOption {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            ReportOption::Nudity => "Nudity/Porn",
+            ReportOption::Violence => "Violence/Gore",
+            ReportOption::Offensive => "Offensive",
+            ReportOption::Spam => "Spam/Ad",
+            ReportOption::Other => "Others",
+        }
+    }
+}
+
+#[server]
+pub async fn send_report_offchain(
+    reporter_id: String,
+    publisher_id: String,
+    publisher_canister_id: String,
+    post_id: String,
+    video_id: String,
+    reason: String,
+    video_url: String,
+) -> Result<(), ServerFnError> {
+    use crate::utils::off_chain;
+    use tonic::metadata::MetadataValue;
+    use tonic::transport::Channel;
+    use tonic::Request;
+
+    let off_chain_agent_url = OFF_CHAIN_AGENT_GRPC_URL.as_ref();
+    let channel = Channel::from_static(off_chain_agent_url).connect().await?;
+
+    let mut off_chain_agent_grpc_auth_token = env::var("GRPC_AUTH_TOKEN").expect("GRPC_AUTH_TOKEN");
```

Comment:
can we migrate off chain client to AppState instead of constructing every time?
https://github.com/go-bazzinga/hot-or-not-web-leptos-ssr/blob/main/src/state/mod.rs

Reply:
done